app/src/features/mappings/MappingsToolbar.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+
+import { Icon } from "@blueprintjs/core";
+import { IconNames } from "@blueprintjs/icons";
+import { Button, Grid, makeStyles, Typography } from "@material-ui/core";
+import AttributeIcon from "@material-ui/icons/LocalOffer";
+import { useParams } from "react-router";
+
+import {
+  useListSourceAttributes,
+  useListSourceResources,
+  useRetrieveSourceQuery,
+} from "services/api/api";
+
+const useStyles = makeStyles((theme) => ({
+  rowContainer: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  button: {
+    textTransform: "none",
+  },
+  icon: {
+    width: 16,
+    height: 16,
+    marginRight: theme.spacing(0.5),
+    fill: theme.palette.text.secondary,
+  },
+}));
+
+const MappingsToolbar = (): JSX.Element => {
+  const classes = useStyles();
+  const { sourceId } = useParams<{ sourceId?: string }>();
```

Comment:
I suggest to use something like `Pick<Partial<Source>, "id">` for the type

Reply:
This doesn't seem to be assignable to string, weird

`Type 'Pick<Partial<Source>, "id">' is not assignable to type 'string'. `